src/app.js

```diff
@@ -11,5 +14,34 @@ riot.install(function (component) {
     installReduxPlugin(component);
 });
 
-riot.register("app", App);
-riot.mount("app");
+const inAppelflap = () => {
+    const { browser, inAppelflap, inPWAMode } = getPlatform();
+    return inAppelflap;
+};
+
+/** Tell Appelflap that Canoe is 'locked'
+ * and should not be rebooted */
+const lockCanoe = async () => {
+    const appelflapConnect = new AppelflapConnect();
+    return await appelflapConnect.lock();
+};
+
+if (inAppelflap()) {
+    lockCanoe()
+        .then((lockResult) => {
+            if (lockResult !== "ok") {
+                // The lock didn't succeed, the app is probably still usable.
+                // We don't yet understand what this means for the user and the app.
+                // Should we show a small warning?
+            }
+
+            riot.register("app", App);
+            riot.mount("app");
+        })
+        .catch((_) => {
+            /* Do nothing */
+        });
+} else {
+    riot.register("app", App);
+    riot.mount("app");
+}
```

Comment:
If it throws I'd rather not swallow it.
Using await avoids the repetition

```suggestion

    const lockResult = await lockCanoe();

    if (lockResult !== "ok") {
        // The lock didn't succeed, the app is probably still usable.
        // We don't yet understand what this means for the user and the app.
        // Should we show a small warning?
    }
}

riot.register("app", App);
riot.mount("app");
```

Reply:
This has all changed.